server/middlewares/validators/videos/videos.ts

```diff
@@ -77,50 +80,123 @@ const videosAddValidator = getCommonVideoEditAttributes().concat([
+ *
+ * Uploadx doesn't use next() until the upload completes, so this middleware has to be placed before uploadx
+ * see https://github.com/kukhariev/node-uploadx/blob/dc9fb4a8ac5a6f481902588e93062f591ec6ef03/packages/core/src/handlers/base-handler.ts#L81-L93
+ *
+ * This validator is meant for POST requests only.
+ */
+/* eslint-enable max-len */
+const videosAddResumableInitValidator = getCommonVideoEditAttributes().concat([
+  body('name')
+    .trim()
+    .custom(isVideoNameValid)
+    .withMessage('Should have a valid name'),
+  body('channelId')
+    .customSanitizer(toIntOrNull)
+    .custom(isIdValid).withMessage('Should have correct video channel id'),
+  header('x-upload-content-length')
+    .isNumeric()
+    .exists()
+    .withMessage('Should specify the file length'),
+  header('x-upload-content-type')
+    .isString()
+    .exists()
+    .withMessage('Should specify the file mimetype'),
+
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+    if (req.body.isPreviewForAudio) return next() // don't check mimetype for audio preview TODO: check size?
+
+    const videoFileMetadata = {
+      mimetype: req.headers['x-upload-content-type'],
+      size: +req.headers['x-upload-content-length'],
+      originalname: req.body.name
+    } as express.FileUploadMetadata
+    const user = res.locals.oauth.token.User
+    const file: UploadxFile & { duration: number, path: string, filename: string } = req.body
+    file.path = getResumableUploadPath(file.id)
+
+    logger.debug('Checking videosAddResumableInitValidator parameters and headers', { parameters: req.body, headers: req.headers })
+
+    if (areValidationErrors(req, res)) return clearUploadFile(file.path)
+    if (areErrorsInScheduleUpdate(req, res)) return clearUploadFile(file.path)
+
+    const cleanup = () => clearUploadFile(file.path)
+
+    /* eslint-disable @typescript-eslint/no-floating-promises */
+    await commonVideoChecks({ req, res, user, videoFileSize: videoFileMetadata.size, file: [ videoFileMetadata ], cleanup })
+
+    try {
+      const storage = new DiskStorage({ directory: getResumableUploadPath() })
+      const userPartiallyUploadedFiles = await storage.get(user.id) // uploadx's default naming relies on req.user.id to prefix file names
+      const userPartiallyUploadedFilesSize = await Promise.all(userPartiallyUploadedFiles.map(f => stat(f.name).then(s => s.size)))
+      const userPartiallyUploadedFilesTotalSize = userPartiallyUploadedFilesSize.reduce((a, b) => a + b, 0) + file.size
+      if (userPartiallyUploadedFilesTotalSize > CONSTRAINTS_FIELDS.VIDEOS.PARTIAL_UPLOAD_SIZE.max) throw new Error()
+    } catch (err) {
+      res.status(HttpStatusCode.PAYLOAD_TOO_LARGE_413)
+         .json({ error: 'This file is too large. It exceeds the partial uploads combined file size authorized.' })
+    }
```

Comment:
@Chocobozzz is it enough of a countermeasure for partial uploads filling the disk?